src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -104,7 +105,8 @@ public EventTypeService(final EventTypeRepository eventTypeRepository,
             eventType.setTopic(topicName);
             eventTypeRepository.saveEventType(eventType);
             return Result.ok();
-        } catch (final InvalidEventTypeException | NoSuchPartitionStrategyException e) {
+        } catch (final InvalidEventTypeException | NoSuchPartitionStrategyException |
+                DuplicatedEventTypeNameException e) {
```

Comment:
I would like to use different approach. The problem there is that it is not safe to do like that in case of runtime exceptions. Please consider the following approach
```
boolean eventTypeSaved = false;
try {
    eventType.setTopic(topicName);
    eventTypeRepository.saveEventType(eventType);
    eventTypeSaved = true;
} finally {
    if (!eventTypeSaved) {
        try {
            topicRepository.deleteTopic(eventType.getTopic());
        } catch (final Exception ex) {
            LOG.error("failed to delete topic for event type that failed to be created", ex);
        }
    }
}
```